### PR TITLE
Clarification of tie-breaker value

### PR DESCRIPTION
**Pull request by cdh4u**
Clarification of tie-breaker value: the same value must be used for all media associated with the ICE session.

OPEN ISSUE: Is it allowed (or, is it mandatory) for an ICE agent to change the tie-breaker value during an ICE restart?

### CODE REVIEW

File: draft-ietf-ice-rfc5245bis.xml [hunk 1]
```diff
@@ -1574,17 +1574,52 @@ protocol enabling the candidate exchange. The state of ICE processing
...
+of an ICE restart (<xref target="sec-restart"/>), but an ICE agent
+MUST NOT re-determine the role as part of an ICE restart unless one
+or more of the following criteria is fulfilled:
+</t>
+
+<t><list style="hanging">
+<t hangText="Full becomes lite:">
+If the controlling agent is full, and switches to lite, the roles MUST be
+re-determined if the peer agent is also full
+</t>
```
Comment (pthatcherg) [inline draft-ietf-ice-rfc5245bis.xml:1592]: Missing a period

File: draft-ietf-ice-rfc5245bis.xml [hunk 2]
```diff
@@ -1574,17 +1574,52 @@ protocol enabling the candidate exchange. The state of ICE processing
...
+
+<t hangText="Role conflict:">
+If the ICE restart causes a role conflict, the roles might be
+re-determined due to the role conflict proceudres in <xref target="sec-role-conflict"/>.
```
Comment (taylor-b) [inline draft-ietf-ice-rfc5245bis.xml]: proceudres -> procedures

File: draft-ietf-ice-rfc5245bis.xml [hunk 3]
```diff
@@ -2157,31 +2197,31 @@ markings to its connectivity checks.
...
 
 <t>
 If the STUN transaction generates a 487 (Role Conflict) error
-response, the agent checks whether it included the ICE-CONTROLLED or
-ICE-CONTROLLING attribute in the Binding request. If the request
-contained the ICE-CONTROLLED attribute, the agent MUST switch to the
+response, the agent checks whether it included an ICE-CONTROLLED or
+ICE-CONTROLLING attribute in the associated Binding request. If the request
+contained an ICE-CONTROLLED attribute, the agent MUST switch to the
 controlling role if it has not already done so. If the request
```
Comment (taylor-b) [inline draft-ietf-ice-rfc5245bis.xml]: Can you also remove "if it has not already done so"? This is one of the things I brought up on the list; it implies you can only switch to a given ICE role one time.

File: draft-ietf-ice-rfc5245bis.xml [hunk 4]
```diff
@@ -2993,17 +3042,21 @@ all media streams using those candidates.
...
-that, during the restart, media can continue to be sent to the
-previously validated pair.  </t>
+<section anchor="sec-restart" title="ICE Restarts">
+<t>
+An agent MAY restart ICE processing for an existing media stream.
+An ICE restart will cause all previous states, excluding the roles
+of the agents, of ICE processing to be flushed and checks to start anew.
+The only difference between an ICE restart and a brand new media session
+is that during the restart media can continue to be sent to the
+previously validated pair, and that a new media session always require
```
Comment (taylor-b) [inline draft-ietf-ice-rfc5245bis.xml]: require->requires

File: draft-ietf-ice-rfc5245bis.xml [hunk 5]
```diff
@@ -4014,18 +4079,26 @@ media. The attribute has no content (the Length field of the attribute
...
-<t>The ICE-CONTROLLED attribute is present in a Binding request and
+<t>
+The ICE-CONTROLLED attribute is present in a Binding request and
 indicates that the client believes it is currently in the controlled
 role. The content of the attribute is a 64-bit unsigned integer in
-network byte order, which contains a random number used for
-tie-breaking of role conflicts.
+network byte order, which contains a random number. The number is used
+for solving role conflicts, when it is referred to as the tie-breaker
+value. An ICE agent MUST use the same number for all Bining requests,
```
Comment (taylor-b) [inline draft-ietf-ice-rfc5245bis.xml]: Bining->Binding

File: draft-ietf-ice-rfc5245bis.xml (hunk 1 above)
Comment (cdh4u) [inline draft-ietf-ice-rfc5245bis.xml:1592]: Will be fixed.

File: draft-ietf-ice-rfc5245bis.xml (hunk 2 above)
Comment (cdh4u) [inline draft-ietf-ice-rfc5245bis.xml]: Will be fixed as suggested.

File: draft-ietf-ice-rfc5245bis.xml (hunk 3 above)
Comment (cdh4u) [inline draft-ietf-ice-rfc5245bis.xml]: Note that there are two instances of "if it has not already done so" in the paragraph. I assume you want to remove both?

File: draft-ietf-ice-rfc5245bis.xml (hunk 4 above)
Comment (cdh4u) [inline draft-ietf-ice-rfc5245bis.xml]: Will be fixed.

File: draft-ietf-ice-rfc5245bis.xml (hunk 5 above)
Comment (cdh4u) [inline draft-ietf-ice-rfc5245bis.xml]: There are two instances of "Bining". Both will be fixed as suggested.

**Comment by pthatcherg** [comment]
FYI, this was merged as https://github.com/ice-wg/rfc5245bis/pull/12/files

**Pull request closed without merging**